### ISSUE TITLE
Discuss analysis of sales data

### ISSUE THREAD

**Issue by jackblun** [huntallcott/Lotteries#15]
I've been looking over the data we have available from Andrew's requests. I've made notes summarizing what they give us [here](https://docs.google.com/document/d/1Wy7SUd-Kix3gwFjrhH_8rsOGkCzFaR0pCGewhtYEM0k/edit#heading=h.i23qpcdct4ok) and identifying the states where the information is potentially useful.

Just to clarify what we want here, the goal of this section of analysis is to estimate:
1. The causal effect of income on lottery purchases
2. The consumer surplus generated by the lottery, and how this varies by income. 

(2) can be identified with data on prices and quantities. In terms of obtaining (1), roughly speaking the ideal experiment would be to elicit valuations for lottery tickets to a set of individuals, randomly vary their income, then elicit valuations again. Given that we cannot do this, we have two options:
i) Use variation in income across geography at a point in time
ii) Use variation in income over geography and time

Approach (i) is done in the literature on lottery regressivity in Oster (2004) and Garrett (2012). They use census data on local incomes and look at how local sales vary by local incomes. This is very feasible given our data, or even using the data from Garrett (2012), but using this would not be consistent with our distinction (in the TESS app) between the observed cross-sectional relationship and the causal relationship.

Therefore we need (ii). To get this we need the following:
a) panel data on ticket sales over time by locality
b) panel data on incomes over time by locality

We can get (a) from the data we have. For the most recent years we might be able to use ACS for (b), but the available geographies will be very aggregated. 

@huntallcott @bblockwood @dtaubinsky , any suggestions on how to proceed or corrections to the above line of argument?

**Comment by huntallcott** [comment]
@jackblun thanks for jumping back into this. 
Yes, I agree with everything you wrote up to the last two paragraphs.
For (b), panel data on incomes, there are several sources. For county-by-year or even county-by-quarter data, there is the REIS or QCEW. We used the REIS in the soda tax paper; I think QCEW has quarterly data, although I don't think that quarterly provides much more useful income variation than annual. We can also go to zip-by-year data from the IRS statistics of income (SOI) data. I also looked at this for the soda tax paper but got some results that I didn't believe as much as with the county data, and so if we went to zips I'd want you to look into those data a bit. 
I can send you pre-written code and data to get (b) for REIS or SOI.

For (a), I had in mind that we would buy the LeFleur's data at the state-by-game-by-year level. My fear with the data @anjoung gathered is that it doesn't go far enough back in time or include enough states to get the precision that we could get from the full national data. I think that ideally we'd have separate estimates of the income effect for scratch-offs as a whole vs. lotto as a whole. 

It seems totally best to see what we get from the data we have. This will give us the code we need, and we can plug in LaFleur's later iff we need it. 
I put the file Prep_zip_Data.do into Lotteries/Code/Build/Geographic, and the SOI and REIS data into dropbox/Externals/Data/Geographic. You might want to check if SOI or REIS has been updated with the next year of data since I downloaded these.

I would suggest that the next step is to run a panel data regression with data at the (zip or county) by year level, regressing ln(expenditures by game in year t) on ln(Income), year FEs, and (zip or county) FEs.

***
I think that income effects are the easy part, though. @jackblun the harder part is that I was also hoping you could dive back into your item (2) above, i.e. the preference estimation (i.e. estimating the utility function parameters as in #9) with the newer data, to see if the estimates are more precise/stable.

*** 
Many thanks on all of this!

**Comment by jackblun** [comment]
@huntallcott thanks for all the above and for clarifying the plan for consumer surplus. From the data we have, estimation of the utility function parameters should be possible with Louisiana, Oregon, and possibly Nebraska (with a lot of work transforming scanned tickets into data), in addition to the estimates we already have from North Carolina.

**Comment by jackblun** [comment]
I have cleaned and analyzed the Oregon data. We are facing the same issues as in North Carolina, in that the nlls method struggles to converge and when it does, estimates are all over the place. All code is pushed to github and should be clear enough. It would be beneficial if someone else had a look through.

I checked the code is working with some dummy data when I did the North Carolina estimates (in the nc code), so I don't think the problem is there. I think the most likely culprits are:
- The sales data is noisy, due to this issue of it being sales to stores rather than to individuals. There were for example negative sales for some games in some years, which are dropped from the current analysis.
- The model is a poor fit to the data

Once I'm back on campus in a couple of weeks I can talk to faculty who are experienced with demand estimation to see whether they can suggest a better way of doing this. One hope is that we could better use the time dimension. Right now for Oregon I am pooling 5 years of sales to get the shares, whereas splitting up and doing by year might make more sense.

Hope this is clear. I will be off-grid in deepest darkest France for the next week.

**Comment by huntallcott** [comment]
@jackblun I hope you had a great time in deepest/darkest France.
I looked through the analysis and had some thoughts.

1. I disagreed with your approach to calculating sales per unit time. The total sales of each ticket are of course determined ex ante and probably have little to do with demand. So we know we want a measure of the flow of sales, somehow normalized for how many stores are offering each ticket. It looks like you aggregate to the year level and then divide total sales by the number of years in which there is at least one sale. This will be noisy for two reasons: (i) you can see in the data that some of these tickets probably run out partway through the year, so for ticket A, there will be 500k sales in one year and 100k sales in the next, and this looks the same in your market share estimates as another ticket B that has 300k sales in one year and 300k in the next, despite the fact that ticket A is probably more popular; and (ii) it seems possible that tickets might be stocked differently, so ticket A might be available at 400 stores, and ticket B at 200 stores, but we aren't accounting for that here.

To deal with issue (i), it seems crucial to get more granular than the yearly level.
To deal with issue (ii), I'd be interested to approach this both by looking at the data and by talking to somebody from the Oregon Lottery Commission. 

Questions:
* If you plot the sales of a bunch of individual tickets across weeks of the data, what does it usually look like? Does it go from zero to 100, stay at 100 for two years, and then drop back to zero when the tickets run out? Or does is ramp up slowly, then ramp down slowly? 
* For each game, determine the number of stores that ever sell that game, and plot the histogram of NStores across games. This tells us how important it is to control for (ii).

Here are two approaches:
A. Right now you do this: `replace net_sales = net_sales/years_sale`. A simple thing you could do is construct game_length = game_end-game_start, then `replace net_sales = net_sales/game_length. But this doesn't address the number of stores selling confound, nor the possibility of a ramp-up and ramp-down.
B. A perhaps better approach would be to look in OregonLottery_Instant_Game_Sales.csv and consider only games that are bought multiple times by the same store. (Important to check how much this limits the sample.) Then you can calculate a sales flow of game j at store s (SalesFlow_js) as net_sales / "time until the next sale". Then we construct our game level sales flow variable (SalesFlow_j) as either the average of SalesFlow_s or the sum of SalesFlow_s. (I lean toward the latter but you could make an argument for the former.)

2. I'm a strong proponent of the idea that any empirical parameter one wants to estimate should be visible in the data. Your [North Carolina writeup](https://docs.google.com/document/d/1Iw7GFYAGwhqU-RN7F-kFpHuS1i1McYLFwsT8bMO2Qbo/edit?usp=sharing) has some interesting figures, but I don't think any of us have figured out how to graphically display the identification of curvature or probability weighting. 
Here is a graph that shows curvature, under linear probability weighting:
regress ln(s_j) = \sum_{k} \beta_k Pr(jk) + \phi_{price} + \varepsilon_j, 
where j indexes games, k indexes payoffs in dollars, Pr(jk) refers to your pi_?? variables, and \phi_{price} is a fixed effect for purchase price ($1, $2, $3, $5, $10, and $20).
Then plot the beta_k and standard errors for the $1 payoffs, the $2 payoffs, etc., all the way to the $1,000,000 payoffs. Presumably these \betas should be increasing.

Can any of you think of a better figure, and/or another figure that could demonstrate probability weighting?

3. Separately from the quantitative discussion, I would be interested to know from a conversation with Oregon Lottery officials, shop owners, and/or Mr. LaFleur, "what makes a scratch-off sell well?"

4. Presumably once we settle on the SalesFlow_s variable, we can and should get total SalesFlow by year and zip for Oregon and the other states in order to look at income effects.

Thoughts?

Thanks again for keeping this ball rolling.

**Comment by huntallcott** [comment]
Also, minor, but please always use saveold instead of save in Stata, as Stata is not backwards compatible and people may not update immediately to the latest version.

**Comment by jackblun** [comment]
Noted on saveold - thanks for the tip. 

I haven't been able to make any progress on the detailed points above as I've just moved back to Stanford and started my RA job. Will get onto your points when I can - apologies.

**Comment by jackblun** [comment]
To respond to the two brief data questions above @huntallcott:

1. I've made a few weekly sales plots for a set of games. For most games, we see a fairly rapid decline in sales over the first few weeks of release. Here are two examples:

![screen shot 2018-10-02 at 17 07 17](https://user-images.githubusercontent.com/13908154/46383564-cd1c9880-c665-11e8-9895-57f98e7d38ec.png)

![screen shot 2018-10-02 at 17 06 56](https://user-images.githubusercontent.com/13908154/46383546-c130d680-c665-11e8-8b60-52c9ab2b90d1.png)

2. Here is the distribution of the number of stores selling each game. We see a few games being sold in fewer than 10 stores, but for the most part the data looks approximately normal:
![screen shot 2018-10-02 at 17 12 43](https://user-images.githubusercontent.com/13908154/46383672-68ae0900-c666-11e8-89c7-8a4127e7e138.png)

**Comment by jackblun** [comment]
Two further graphs that may be of interest @huntallcott . Firstly the number of games in each store in each week:
![screen shot 2018-10-03 at 12 03 14](https://user-images.githubusercontent.com/13908154/46432951-65695a80-c704-11e8-8812-cc0c1ed9998c.png)
Also the total number of games  each week across all stores:
![screen shot 2018-10-03 at 12 04 28](https://user-images.githubusercontent.com/13908154/46432989-85991980-c704-11e8-8b3f-7f8902373ac7.png)
:

**Comment by huntallcott** [comment]
@jackblun thanks for this. 

1. The first of these two posts suggests to me that my approach A [above](https://github.com/huntallcott/Lotteries/issues/15#issuecomment-422522610) won't work so well, so we'd have to go with approach B. But for approach B I'm also more worried about, on what share of purchases can you construct the SalesFlow variable I proposed. i.e. what share of sales have another subsequent sale of the same game at the same store? If that share is pretty low, then we will have only a noisy sense of the flow of demand.
It could be that we need to go back to states such as North Carolina (e.g. [here](https://www.nclottery.com/ScratchOffDetail?gn=622)) where we can get a good sense of the rate of sales to *end* consumers, not the stores.
On the other hand, it would be good to know whether customers shop around across stores or just have one favorite store and buy their favorite ticket in that store. If the latter, then your third graph (out of the most recent four above), if I'm reading it correctly, suggests that the choice set will vary across stores, and that could be used for identification.

2. I didn't quite understand what was happening in the second pair of graphs. What exactly do you mean “games in each store in each week”? We don’t know if a game has sold out or not, so we don’t know if it’s actually “in” the store, right?

**Comment by jackblun** [comment]
1. I agree this strategy is difficult with the data on sales to stores for Oregon. I've followed up with the Oregon people about how these sales figures are put together and am awaiting reply. It may be they have something more disaggregated. Other than this feature, the Oregon data has a lot of what we need.

2. In these graphs "games in each store in each week" corresponds to the number of different games ordered by the store in each week. It doesn't correspond directly to the number of games actually on sale that week, just those which have been ordered in. 

The second graph then shows the total number of games ordered each week by stores. As you point out though, given that the stores seem to order a bunch of stock when the games are first released, this may not be that informative for number of games on sale to consumers in a given week, its more just to understand patterns in this dataset.

**Comment by jackblun** [comment]
Correct on your first point - lets hope Oregon has something better for us. I'll go through the other states' data again.

My instinct with the LeFleurs is that they are quite helpful if they think we will buy something..! I haven't spoken to them in a little while. I think if we have some specific questions which are also related to their data holdings, a call could be useful. Do you have an idea of exactly what we would hope to get out of them?

**Comment by huntallcott** [comment]
@jackblun the basic question in my mind would be what states have the right data for this exercise. We could pitch it as a combo of interest in their data vs. understanding the broader data landscape. It could be that @anjoung's work was the best way to answer that question, but it could also definitely be that 30 minutes on the phone with him saves us weeks of work. But if he's grumpy or focused only on his own data and sales, then it would be less productive. Maybe if and only if you think it would be good, you could try to schedule a conversation with him one afternoon next week?

**Comment by jackblun** [comment]
@huntallcott agreed. I'll organize a call next week - I'll let you know the times Byron LaFleur is free.

**Comment by jackblun** [comment]
I've gone over the other states' data. For retailer-level sales, the situation is the same as it is for Oregon. We have data on sales to stores for a given time period, which is essentially just lumpy stock orders and not necessarily reflective of actual sales to consumers in the period. I think we will either need other data or an alternative strategy.

**Comment by jackblun** [comment]
@huntallcott I've had a look through the Dimmock et al paper you sent over. The approach used there would be another option for us - estimating the utility parameters (curvature and probability weighting) through a survey module rather than using actual market data. I can see this is less attractive as choices may not reflect those in real-life lottery markets, and it would require a bunch more questions in our survey. Do you have any thoughts on this as an option?

**Comment by huntallcott** [comment]
Thanks Jack.
Let’s see what lefleur says.
The nc data should still be viable, yes?
Our other option to do this with market data is to go back to estimating
with lotto sales. There we should have actual purchases.
-- 

Hunt Allcott

https://sites.google.com/site/allcott/home

**Comment by huntallcott** [comment]
@jackblundell, sorry for the delay.
This is a cool idea, and I like that you've got your eyes open for things
like that. My intuition is that this should be a fallback only, for
precisely the reason that you stated -- we want to shoot to use market data
whenever possible, due to concerns about external validity.



-- 

Hunt Allcott

https://sites.google.com/site/allcott/home

**Comment by jackblun** [comment]
Response from Oregon suggests data is even worse than we thought. They do conveniently note however that monthly aggregation seems to approximate true sales reasonably well, so that would be something to try:
- Activations is the face value of the tickets that are removed from their pack and made available for sale.  Tickets come in packs and are considered to be 'activated' when the pack is opened. Activations will be in multiples of 100.
- Returns is the face value of the tickets that the retailer doesn't sell and returns to the Lottery for destruction. This usually happens after a game is discontinued or a retailer closes.
- Adjustments are changes due to promotions, free tickets, etc.
- Net Sales is equal to Activations + Returns + Adjustments.  We can't actually measure true sales because we do not get data from each retailer's registers.  Instead, we approximate it by taking the value of the tickets that are activated and subtracting the value of the returns.  It's important to note that daily sales are likely misleading since a retailer may activate hundreds of dollars of tickets on a particular day and then not activate anything else for another week or so.  This doesn't mean that the retailer isn't selling any tickets; just that the retailer isn't activating tickets. Monthly aggregations of net sales seems to approximate true sales fairly well.

**Comment by huntallcott** [comment]
@jackblun: 
A. It seems like it's still worth understanding stores' ordering and activation strategies, in order to know whether my strategy B [here](https://github.com/huntallcott/Lotteries/issues/15#issuecomment-422522610) will work. Possibility (i) is that stores open packs for specific games exactly when the previous pack for that specific game ran out. Possibility (ii) is that stores order and activate in groups, e.g. when you feel like you are low on all tickets, you make a big order, and then when that order arrives, you open (activate) all the packs at once and add them to the shelf. If (i) is the case, then strategy B works. If (ii) is the case, then maybe not. The patterns in your first graph [here](https://github.com/huntallcott/Lotteries/issues/15#issuecomment-426761058) (which is activations, yes?) made me think we are in case (ii). Do you have a sense here?
B. Given A and other things you know about the data, would you believe my strategy B [here](https://github.com/huntallcott/Lotteries/issues/15#issuecomment-422522610)? 

Most broadly, I presume that we're on the same page about what the key issues are here? Are you able to pursue the people from Oregon and other states to really understand what's in the data and whether it will work? This issue seems subtle enough that we can't give it to a pre-doc, even the very best pre-docs.

**Comment by jackblun** [comment]
A. The graphs of activations over time suggest (ii) is the case to me, that suppliers make a big order, activate everything and then sell as much as they can, returning the remainder to the state when the game is over. 

B. My feeling is that this issue with activations will render B not possible with the Oregon data. 

On the broader comment, this afternoon I am meeting with a couple of IO guys here to discuss the empirical strategy. I think this will help me understand exactly what we need from the data, and where we can make do with sub-optimal data. Then I intend on following up directly with a few more states to see if they can give us anything closer to the optimum than they did last time.

**Comment by jackblun** [comment]
Judging from the data, it is the case that the game ends at some fixed date and unsold tickets are returned at that date. However in practice there aren't that many returns, so the date is set far out enough relative to the total number of tickets that the suggested method may not work.

I discussed the empirical problem in general with Martin O'Connell, who you may have spoken to about soda taxes since he's done some work on that, but from a very IO angle. Discussing with him makes me think we should also re-consider using the state-level lotto (rather than scratch-off) sales data for this part of the paper. He thinks the approach is reasonable.

**Comment by huntallcott** [comment]
Thanks.

Quoting from our meeting notes 7-3-2018, we liked scratch-offs for the following four reasons:
1. Lots of purchases of scratchoffs.  63% of U.S. lotteries’ total traditional sales in 2017. Lotto+keno is about 37%. (excludes non-traditional: video lottery, casino games, sports betting). It’s 50-60% when you include the non-traditional.
2. Temptation might be a larger share of bias, and we know how to estimate that.
3. Identification could be more credible because the lotto identification would require assuming away dynamics (or addressing them in a complex way)
4. Because some of the payoffs are lower, we may be able to credibly incorporate a price elasticity wrt the scratch-off payment.

(1) is not the end of the world.
(2) actually speaks to lotto now given that our bias measurement on TESS isn't focused on temptation. but I do fear we'd be missing something key here -- temptation seems very relevant given that there are so many sales of instant games. maybe it's fine to say that our paper is just about lotto games.
(3) I think we can deal with this or get away with assuming it away.
(4) This is a problem. One thing we could do is identify price elasticity from the TESS MPL by testing how much demand increases when we drop price below $1 per ticket. (Price above $1 per ticket is less credible given the possibility for making purchases outside the experiment.)

I also note that using the LaFleur's data, we can estimate substitution patterns to other games in response to weekly variation in the lotto jackpots. That seems interesting.

In light of this, my questions for Bryron LaFleur would be:
1. Are there any states where can get good scratch-off consumer demand data, as in North Carolina?
2. Does he have store-level sales over time, or just state-level? (Understanding demand by income is important for determining optimal payout structure by income.) (@jackblun am I right that the store-level data we gathered so far from states are just for scratch-offs?) 
3. Can we get a discount on his data?

@bblockwood @dtaubinsky what do you think about re-focusing the demand estimation on lotto tickets?

**Comment by jackblun** [comment]
Thanks for the clarity above @huntallcott . You are right that our store-level data at the moment is just scratchoffs. On (4), we do potentially have the shift in price of Mega millions to estimate a price elasticity, but perhaps that is not enough.

Agreed on questions. I'm still waiting to hear back from Byron - he said he had an emergency yesterday so I won't pester him over the weekend.

**Comment by jackblun** [comment]
Notes from call with LaFleur. Remember to caveat all this with the fact that they would like to sell us their data and not push us towards any alternatives. I tried to emphasise how any additional data would be complementary to theirs rather than substitutes, nonetheless keep this in mind.

1. They are not aware of any store-level datasets, and were actually a little surprised that we had obtained some store-level data from States.
2. They are also not aware of any data containing detail on actual payouts and probabilities for scratchoffs.
3. On price, it was difficult to ask for flexibility since we don't know which of their products (sets of years) we are asking for.
4. They caution against looking at any single type of lottery (eg just instants) since their sales are all interrelated. This could be a problem for us if we look only at either lotto or scratchoffs as planned.
5. Something worth noting is that they do not like the Clotfelder studies one bit! 

All in all, no great breakthroughs. We have actually found more data than they realized was available. They are very reachable by phone if we want to discuss more.
